python/plugins/processing/algs/qgis/Clip.py

```diff
@@ -84,7 +84,7 @@ def processAlgorithm(self, parameters, context, feedback):
 
         # first build up a list of clip geometries
         clip_geoms = []
-        for mask_feature in mask_source.getFeatures(QgsFeatureRequest().setSubsetOfAttributes([])):
+        for mask_feature in mask_source.getFeatures(QgsFeatureRequest().setSubsetOfAttributes([]).setDestinationCrs(feature_source.sourceCrs())):
```

Comment:
So it means we can do a clipping with 2 layers with different CRS ? Nice. It has to be written in the help of the algorithm.